src/pkg/egress/syslog/writer_factory.go

```diff
@@ -41,26 +41,30 @@ type WriterFactory struct {
 	externalTlsConfig *tls.Config
 	netConf           NetworkTimeoutConfig
 	m                 metricClient
+	emitter           AppLogEmitter
 }
 
-func NewWriterFactory(internalTlsConfig *tls.Config, externalTlsConfig *tls.Config, netConf NetworkTimeoutConfig, m metricClient) WriterFactory {
+func NewWriterFactory(internalTlsConfig *tls.Config, externalTlsConfig *tls.Config, netConf NetworkTimeoutConfig, m metricClient, emitter AppLogEmitter) WriterFactory {
 	return WriterFactory{
 		internalTlsConfig: internalTlsConfig,
 		externalTlsConfig: externalTlsConfig,
 		netConf:           netConf,
 		m:                 m,
+		emitter:           emitter,
 	}
 }
 
-func (f WriterFactory) NewWriter(ub *URLBinding) (egress.WriteCloser, error) {
+func (f WriterFactory) NewWriter(ub *URLBinding, emitter AppLogEmitter) (egress.WriteCloser, error) {
```

Comment:
What is the utility of providing `WriterFactory` with an `AppLogEmitter` and passing another `AppLogEmitter` to `WriterFactory.NewWriter()`? It seems to me like they are the same `AppLogEmitter` and so could be de-duped, but maybe I'm missing something.